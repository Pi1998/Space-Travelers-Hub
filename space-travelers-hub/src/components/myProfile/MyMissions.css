.title {
  margin-bottom: 20px;
  font-size: 1.8rem;
  font-weight: 500;
}

.missions-container {
  margin: 20px;
  width: 50%;
}

.missions-list {
  width: 100%;
  border-radius: 7px;
  border: 0.5px solid rgb(72, 72, 72);
  list-style: none;
}

.mission {
  border-bottom: 0.5px solid rgb(72, 72, 72);
  padding: 20px;
  font-size: 1.2rem;
  font-weight: 400;
}

.mission:last-child {
  border: none;
}
