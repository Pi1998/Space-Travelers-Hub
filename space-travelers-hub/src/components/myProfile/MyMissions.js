import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissions } from '../../missions/missionsSlice';

const MyMissions = () => {
  const mission = useSelector((state) => state.mission);
  const dispatch = useDispatch();
  useEffect(() => {
    if (mission.missions.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch, mission.missions.length]);
  const { missions } = mission;
  const myMissions = missions.filter((mission) => mission.mission_join === true);
  return (
    <div className="missions-container">
      <h2 className="title">My Missoins</h2>
      <ul className="missions-list">
        {myMissions.map((mission) => (
          <li className="mission" key={mission.mission_id}>{mission.mission_name}</li>
        ))}
      </ul>
    </div>
  );
};

export default MyMissions;
