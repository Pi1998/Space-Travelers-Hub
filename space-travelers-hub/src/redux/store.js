import { configureStore } from '@reduxjs/toolkit';
<<<<<<< HEAD
import missionsReducer from '../missions/missionsSlice';

const store = configureStore({
  reducer: {
    mission: missionsReducer,
  },
});
=======
import rocketsReducer from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

>>>>>>> 098d084507bd426f6df3ec71c9f14c6ebc81642e
export default store;
