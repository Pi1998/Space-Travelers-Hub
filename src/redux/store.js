import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    mission: missionsReducer,
    rockets: rocketsReducer,
  },
});

export default store;
