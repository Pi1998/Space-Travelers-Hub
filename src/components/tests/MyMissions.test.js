import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyMissions from '../myProfile/MyMissions';
import store from '../../redux/store';

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MyMissions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
