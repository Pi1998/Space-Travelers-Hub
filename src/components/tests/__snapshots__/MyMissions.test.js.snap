// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="missions-container"
>
  <h2
    className="title"
  >
    My Missoins
  </h2>
  <ul
    className="missions-list"
  />
</div>
`;
